tools/tpm2_import.c

```diff
@@ -366,30 +386,50 @@ static int check_options(void) {
 
     int rc = 0;
 
-    if (!ctx.input_key_file) {
-        LOG_ERR("Expected to be imported key data to be specified via \"-i\","
-                " missing option.");
-        rc = -1;
+    if(ctx.input_seed_file == NULL) {
+        if (!ctx.input_key_file) {
+            LOG_ERR("Expected to be imported key data to be specified via \"-i\","
+                    " missing option.");
+            rc = -1;
+        }
+        if (!ctx.key_type) {
+            LOG_ERR("Expected key type to be specified via \"-G\","
+                    " missing option.");
+            rc = -1;
+        }
+    } else {
+        /* Algorithm */
+        ctx.input_enc_key_type = tpm2_alg_util_from_optarg(ctx.name_alg, 
+                tpm2_alg_util_flags_symmetric | tpm2_alg_util_flags_misc);
+        switch(ctx.input_enc_key_type) {
+            case TPM2_ALG_ERROR:
+                LOG_ERR("Unsupported sym encryption key type");
+                rc = -1;
+                break;
+            case TPM2_ALG_NULL:
+                break;
+            default:
+                if(ctx.input_enc_key_file == NULL) {
+                    LOG_ERR("Expected a symmetric encryption key file to be specified via \"-k\","
+                            " missing option.");
+                    rc = -1;
+                }
+                break;
+        }
```

Comment:
this hunk is ultra confusing because of the overloaded -g. If you can re-use -i for the input dup file and -G to say, hey were duplicating it, it should become simpler to read.

Reply:
Will revisit this

Reply:
Agreed, it does make things a lot simpler _but_ this needs to change - the  tpm2_alg_util_flags_* are wrong

    case 'G':
        ctx.key_type = tpm2_alg_util_from_optarg(value,
                tpm2_alg_util_flags_asymmetric
                |tpm2_alg_util_flags_symmetric);
        if (ctx.key_type == TPM2_ALG_ERROR) {
            LOG_ERR("Unsupported key type");
            return false;
        }